src/scripts/main.js

```diff
@@ -1 +1,18 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const gallery = document.querySelector('.gallery');
+  const mainImage = document.querySelector('.main-image img');
+
+  gallery.addEventListener('click', (e) => {
+    let target = e.target;
+
+    if (target.tagName === 'A') {
+      target = target.querySelector('img');
```

Comment:
The logic here assumes that the anchor tag always contains an image. If there are anchor tags without images, this could lead to unexpected behavior. Consider adding a check to ensure that an image exists within the anchor before attempting to select it.